presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryManager.java

```diff
@@ -195,11 +195,17 @@ public synchronized void process(Iterable<QueryExecution> queries)
                     .collect(toImmutableList());
             Optional<QueryId> chosenQueryId = lowMemoryKiller.chooseQueryToKill(queryMemoryInfoList, nodeMemoryInfos);
             if (chosenQueryId.isPresent()) {
-                QueryExecution chosenQuery = Streams.stream(queries).filter(query -> chosenQueryId.get().equals(query.getQueryId())).collect(onlyElement());
-                chosenQuery.fail(new PrestoException(CLUSTER_OUT_OF_MEMORY, "Query killed because the cluster is out of memory. Please try again in a few minutes."));
-                queriesKilledDueToOutOfMemory.incrementAndGet();
-                lastKilledQuery = chosenQueryId.get();
-                logQueryKill(chosenQueryId.get(), nodeMemoryInfos);
+                Optional<QueryExecution> chosenQuery = Streams.stream(queries).filter(query -> chosenQueryId.get().equals(query.getQueryId())).collect(toOptional());
+                if (chosenQuery.isPresent()) {
+                    chosenQuery.get().fail(new PrestoException(CLUSTER_OUT_OF_MEMORY, "Query killed because the cluster is out of memory. Please try again in a few minutes."));
+                    queriesKilledDueToOutOfMemory.incrementAndGet();
+                    lastKilledQuery = chosenQueryId.get();
+                    logQueryKill(chosenQueryId.get(), nodeMemoryInfos);
+                }
+                else {
+                    // See isLastKilledQueryGone
+                    log.info("Query Kill: A non-existent query id is chosen: %s", chosenQueryId.get());
```

Comment:
Nit: capitalize ID

Is this a bug or just a race? If it's a bug, probably better to make this a warning or error